Model/AdyenOrderPaymentStatus.php

```diff
@@ -55,37 +55,30 @@ public function __construct(
         \Magento\Sales\Api\OrderRepositoryInterface $orderRepository,
         \Adyen\Payment\Logger\AdyenLogger $adyenLogger,
         \Adyen\Payment\Helper\Data $adyenHelper
-    )
-    {
+    ) {
         $this->orderRepository = $orderRepository;
         $this->adyenLogger = $adyenLogger;
         $this->adyenHelper = $adyenHelper;
     }
 
     /**
-     * {@inheritDoc}
-     * @throws MagentoNoSuchEntityException
-     * @throws AdyenException
+     * @param string $orderId
+     * @return bool|string
      */
     public function getOrderPaymentStatus($orderId)
     {
-        try {
-            $order = $this->orderRepository->get($orderId);
-            $payment = $order->getPayment();
+        $order = $this->orderRepository->get($orderId);
+        $payment = $order->getPayment();
 
-            if ($payment->getMethod() === AdyenCcConfigProvider::CODE) {
-                return $this->adyenHelper->buildThreeDS2ProcessResponseJson(
-                    $payment->getAdditionalInformation('threeDSType'),
-                    $payment->getAdditionalInformation('threeDS2Token')
-                );
-            } else {
-                return $result;
-            }
-        } catch (NoSuchEntityException $e) {
-            $this->adyenLogger->error("Exception: " . $e->getMessage());
-            throw new \Magento\Framework\Exception\LocalizedException(__('This order no longer exists.'));
+        if ($payment->getMethod() === AdyenCcConfigProvider::CODE ||
+            $payment->getMethod() === AdyenOneclickConfigProvider::CODE
+        ) {
+            $additionalInformation = $payment->getAdditionalInformation();
+            return $this->adyenHelper->buildThreeDS2ProcessResponseJson(
+                $additionalInformation['threeDSType'],
+                $additionalInformation['threeDS2Token']
+            );
         }
```

Comment:
$this->orderRepository->get($orderId); can still throw an exception if the orderId supplied does not exist. Shouldn't we keep the try/catch block to handle it?

Reply:
@acampos1916 I can indeed but the front-end is not able to catch this exception as this time and as well quite a lot of work to implement as it should never happen. I believe this should give a server error so the merchant during development will notice that they probably build a plugin that changed the response causing this to fail. I am open for any other suggestion but believe most of them will need quite a lot of development for it to make sure the front-end will understand this error.